packages/grafana-ui/src/components/FieldConfigs/FieldConfigEditor.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import {
+  FieldConfigEditorRegistry,
+  FieldConfigSource,
+  DataFrame,
+  FieldPropertyEditorItem,
+  GrafanaTheme,
+} from '@grafana/data';
+import { standardFieldConfigEditorRegistry } from './standardFieldConfigEditorRegistry';
+import { stylesFactory } from '../../themes';
+import { css } from 'emotion';
+import { Themeable } from '../../types';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  fieldEditor: css`
+    border: 1px solid red;
+  `,
+  customEditor: css`
+    border: 1px solid green;
+  `,
+}));
+
+interface Props extends Themeable {
+  config: FieldConfigSource;
+  custom?: FieldConfigEditorRegistry; // custom fields
+  include?: string[]; // Ordered list of which fields should be shown/included
+  onChange: (config: FieldConfigSource) => void;
+
+  // Helpful for IntelliSense
+  data: DataFrame[];
+}
+
+interface State {
+  //
+}
+
+/**
+ * Expects the container div to have size set and will fill it 100%
+ */
+export class FieldConfigEditor extends React.PureComponent<Props, State> {
+  styles: any; // ???
+
+  constructor(props: Props) {
+    super(props);
+
+    this.styles = getStyles(this.props.theme);
+  }
+
+  private setDefaultValue = (name: string, value: any, custom: boolean) => {
+    const defaults = { ...this.props.config.defaults };
+    const remove = value === undefined || value === null;
+    if (custom) {
+      if (defaults.custom) {
+        if (remove) {
+          defaults.custom = { ...defaults.custom }; // TODO!!
+          delete defaults.custom[name]; // something better???
+        } else {
+          defaults.custom = { ...defaults.custom, [name]: value };
+        }
+      } else if (remove) {
+        delete (defaults as any)[name];
+      } else {
+        (defaults as any)[name] = value;
+      }
+    }
+    this.props.onChange({
+      ...this.props.config,
+      defaults,
+    });
+  };
+
+  renderEditor(item: FieldPropertyEditorItem, custom: boolean) {
+    const config = this.props.config.defaults;
+    const value = custom ? (config.custom ? config.custom[item.id] : undefined) : (config as any)[item.id];
```

Comment:
What are/will be the customs in FieldConfig used for, can you remind me @ryantxu?

Reply:
Are those, that's custom overrides, right?